nbs/explains/background_tasks.ipynb

```diff
@@ -0,0 +1,296 @@
+    "5. Call the slow API service (simulating a time-consuming operation)\n",
+    "6. Store both timestamps in our database\n",
+    "7. Create a background task by passing in the function to a BackgroundTask object, followed by any arguments.\n",
+    "8. In FtResponse, use the background keyword argument to set the task to be run after the HTTP response is generated.\n",
+    "9. Endpoint that displays all recorded timestamp pairs.\n",
+    "10. When this button is pressed, the 'submit' handler will respond instantly. The task_submit function will insert the slow API response into the db later. \n",
+    "11. Every 5 seconds get the tstamps stored in the DB."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "::: {.callout-tip}\n",
+    "\n",
+    "In the example above we use a synchronous background task function set in the `FtResponse` of a synchronous handler. However, we can also use asynchronous functions and handlers.\n",
+    "\n",
+    ":::"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Multiple background tasks in a handler"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It is possible to add multiple background tasks to an FtResponse."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "::: {.callout-warning}\n",
+    "Multiple background tasks on a background task are executed in order. In the case a task raises an exception, following tasks will not get the opportunity to be executed.\n",
+    ":::"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "from fasthtml.common import *\n",
+    "from starlette.background import BackgroundTasks\n",
+    "\n",
+    "@rt\n",
+    "async def signup(email, username):\n",
+    "    tasks = BackgroundTasks()\n",
+    "    tasks.add_task(send_welcome_email, to_address=email)\n",
+    "    tasks.add_task(send_admin_notification, username=username)\n",
+    "    cts = Titled('Signup successful!')\n",
+    "    return FtResponse(cts, background=tasks)\n",
+    "\n",
+    "async def send_welcome_email(to_address):\n",
+    "    ...\n",
+    "\n",
+    "async def send_admin_notification(username):\n",
+    "    ...\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Background tasks are not distributed task queues\n",
+    "\n",
+    "While background tasks often provide the user with a faster experience, the server itself isn't accelerated. This means processes are still happening, but are hidden from the user. So if a server is struggling under the load of a lot of user activity, so long as it isn't an issue with open HTTP connections consuming resources, background tasks won't help with server load.\n",
```

Comment:
That's not necessarily true AFAICT. E.g if the server is struggling because a non-async process is blocking, then putting it in a bg task will help. So maybe worth mentioning the benefit that bg tasks are async would help here? And perhaps that spinning out a *process* manually can be a useful step between bg tasks and task queues, since it allows us to use multiple cores and avoid the GIL whilst still doing everything in fasthtml?